pandas/io/excel/_openpyxl.py

```diff
@@ -448,52 +487,123 @@ def write_cells(
+
+            for cell in cells:
+                xval, fmt = self._value_with_fmt(cell.val)
+                xcell = WriteOnlyCell(wks, value=xval)
+
+                if fmt:
+                    xcell.number_format = fmt
+
+                if cell.style:
+                    key = str(cell.style)
+                    style_kwargs = _style_cache.get(key)
+                    if style_kwargs is None:
+                        style_kwargs = self._convert_to_style_kwargs(cell.style)
+                        _style_cache[key] = style_kwargs
+
+                # This is primarily the issue. Setting cell attributes arbitrarily.
+                # Might not actually be an issue, as long as we set THEN append
+                if style_kwargs:
+                    for k, v in style_kwargs.items():
+                        setattr(xcell, k, v)
+
+                """
+
+                The way that merged cells work in write_only worksheets is as follows:
+
+                You append the data you want first, then you append the merge over it.
+                It must be in excel notation (e.g. "A1:C1") rather than by index
+                (e.g. 11:14)
+
+                I think creating a helper function to turn the indexes into excel
+                notation could be useful
+
+                """
+                if cell.mergestart is not None and cell.mergeend is not None:
+
+                    wks.merged_cells.ranges.append(
+                        "%s:%s" % (
+                            self._convert_to_excel_format(
+                                row=startrow + cell.row + 1,
+                                col=startcol + cell.col + 1),
+
+                            self._convert_to_excel_format(
+                                row=startrow + cell.mergestart + 1,
+                                col=startcol + cell.mergeend + 1
+                            )
+                        )
+                    )
+
+                    # When cells are merged only the top-left cell is preserved
+                    # The behaviour of the other cells in a merged range is
+                    # undefined
+                    if style_kwargs:
+                        first_row = startrow + cell.row + 1
+                        last_row = startrow + cell.mergestart + 1
+                        first_col = startcol + cell.col + 1
+                        last_col = startcol + cell.mergeend + 1
+
+                        for row in range(first_row, last_row + 1):
+                            for col in range(first_col, last_col + 1):
+                                if row == first_row and col == first_col:
+                                    # Ignore first cell. It is already handled.
+                                    continue
+                                xcell = wks.cell(column=col, row=row)
+                                for k, v in style_kwargs.items():
+                                    setattr(xcell, k, v)
```

Comment:
Haven't finished this part yet. I'm still trying to figure out a way to apply the style to the other cells in the merged range. I'm not entirely sure there is one, but I'll spend more time prodding around.

If I can't find a reasonable workaround, does this have serious ramifications? I imagine probably not given that it's just styling the underlying cells, but maybe I'm missing something.